apps/remote_control/lib/lexical/remote_control/search/fuzzy/scorer.ex

```diff
@@ -0,0 +1,193 @@
+defmodule Lexical.RemoteControl.Search.Fuzzy.Scorer do
+  @moduledoc """
+  Scores a match based on heuristics.
+
+  The goal of this module is to have a quick to build and fast to query fuzzy matching system.
+
+  Matches match a subject based on a pattern given by the user. A string is considered to match
+  if the subject contains all of the pattern's characters, in any order. However, patterns
+  can be boosted based on heuristics.
+
+  The heuristics boost:
+    1. Larger patterns
+    2. Patterns that match more consecutive characters
+    3. Patterns that match the beginning of the subject
+    4. Patterns that match the case of the subject
+
+  Based loosely on https://medium.com/@Srekel/implementing-a-fuzzy-search-algorithm-for-the-debuginator-cacc349e6c55
+  """
+  defstruct match?: false,
+            index: 0,
+            matched_character_positions: []
+
+  import Record
+
+  defrecord :subject, original: nil, graphemes: nil, normalized: nil
+
+  @typedoc "A match score. Higher numbers mean a more relevant match."
+  @type score :: integer
+  @type score_result :: {match? :: boolean(), score}
+  @type subject :: term()
+  @type pattern :: String.t()
+  @type preprocessed ::
+          record(:subject, original: String.t(), graphemes: tuple(), normalized: String.t())
+  @non_match_score -500
+
+  @doc """
+  Pre-processes a subject into separate parts that will be helpful during the search phase.
+  Pre-processing allows us to do the work of extracting important metadata per-subject
+  rather than on every request.
+  """
+  @spec preprocess(subject()) :: preprocessed()
+  def preprocess(subject) when is_binary(subject) do
+    graphemes =
+      subject
+      |> String.graphemes()
+      |> List.to_tuple()
+
+    subject(original: subject, graphemes: graphemes, normalized: normalize(subject))
+  end
+
+  def preprocess(subject) do
+    subject
+    |> inspect()
```

Comment:
```suggestion
    |> stringify()
```
But we should be using the function passed in to fuzzy. Maybe scorer should only accept strings as subjects